### PR TITLE
Test Driven Development

### PR DESCRIPTION
**Pull request by Kccendana**
### **Project requirements**

- Create a class called Solver.
- Create a method called factorial that takes one argument, an integer N, and returns the factorial for that number. The factorial is the multiplication of all integers from 1 to N and has the special case that the factorial of 0 is 1. This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.
- Create a method called reverse that takes one argument, a string word, and returns word reversed (e.g. if word is "hello" it returns "olleh").
- Create a method called fizzbuzz that takes one argument, an integer N, and returns a string. The returned string is constructed following these rules:
  - When N is divisible by 3, return "fizz".
  - When N is divisible by 5, return "buzz".
  - When N is divisible by 3 and 5, return "fizzbuzz".
 - In any other case, return N as a string (e.g. say N is 7 then return "7").
- Remember to start with tests for all of the methods described above. Your commit history should prove that you have used TDD.

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+require_relative '../solver'
+
+describe Solver do
+  solver = Solver.new
+
+  context '#factorial' do
+    it 'returns the factorial of the number' do
+      expect(solver.factorial(5)).to eql(120)
+    end
+  end
```
Comment (omar-muhamad) [inline spec/solver_spec.rb]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/testing/tdd_project.md#:~:text=to%20N%20and-,has%20the%20special%20case%20that%20the%20factorial%20of%200%20is%201.%20This%20method%20only%20accepts%200%20and%20positive%20integers%2C%20so%20if%20a%20negative%20integer%20is%20given%20it%20should%20raise%20an%20exception.,-Create%20a%20method), Please make sure that you cover all edge cases inside your tests: 
- if a negative integer is given it should raise an exception.
- special case that the factorial of 0 is 1.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @Kccendana and @chudisoft,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by shahadat3669 — APPROVED** [review]
Hi everyone, 

Your project is complete!  There is nothing else to say other than it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ cover all edge cases
✅ checked special case that the factorial of 0 is 1.
✅ Has class with name Solver implementing #factorial, #reverse and #fizzbuzz methods

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table  width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadata3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
